xbmc/cores/VideoPlayer/DVDCodecs/Video/VDPAU.cpp

```diff
@@ -330,6 +332,24 @@ bool CVDPAUContext::Supports(VdpVideoMixerFeature feature)
   return false;
 }
 
+bool CVDPAUContext::IsValidDecoder(CDecoder *decoder)
+{
+  auto it = find(m_decoders.begin(), m_decoders.end(), decoder);
+  if (it != m_decoders.end())
+    return true;
+
+  return false;
+}
+
+void CVDPAUContext::FFReleaseBuffer(void *opaque, uint8_t *data)
+{
+  CDecoder *vdp = (CDecoder*)opaque;
```

Comment:
reinterpret_cast is for nasty type conversions which seems not appropriate here. Here we convert a pointer to a pointer of a specific type -> static_cast